### PR TITLE
Total Refactor, a New Hope

### PR DESCRIPTION
**Pull request by Kikobeats**
## Basic

- [x] Added npm scripts instead of Makefile.
- [x] Integrated with [standard](https://github.com/feross/standard) linter.
- [x] Integrated tests with mocha & istanbul.
- [x] Reorganized examples.
- [x] Fixed linter issues.
- [x] Fixed fails tests.
- [x] Integrated PR builds with travis and coveralls.

## Improvements

- [x] Extract `node.stream()` method instead of `node.write()` without arguments.

### CODE REVIEW

**Comment by Kikobeats** [comment]
Please join in travis-ci.org and activate this repository to enable travis on PR. Do the same for https://coveralls.io/ as well.

**Comment by Kikobeats** [comment]
Need help to resolve this linter problems:

```
// create the stream
        stream = stream
          ? stream
          : paginate
            ? stream_array(state.stream)
            : stream_object(state.stream)
```

message: `Unnecessary use of conditional expression for default assignment`

-

```
if (src = $el.attr('href')) {
      key = 'href'
    } else if (src = $el.attr('src')) {
      key = 'src'
    } else {
      return
    }
```

message: `Expected a conditional expression and instead saw an assignment.` (x2)

**Comment by Kikobeats** [comment]
Phantom test still broken:

```
driver(fn)
  x-ray params: {"source":"http://www.google.com/ncr","scope":null,"selector":"title"} +4ms
  x-ray starting at: http://www.google.com/ncr +0ms
  x-ray fetching http://www.google.com/ncr +0ms
phantom stdout: NETWORK_ERR: XMLHttpRequest Exception 101: A network error occurred in synchronous requests.


phantom stdout:   phantomjs://code/shim.js:1815 in send
  phantomjs://code/shim.js:1815 in _start
  phantomjs://code/shim.js:1842
```

**Comment by matthewmueller** [comment]
+1 looks good :-)

i've gone ahead and given you commit access to this project. for big things, please open a PR. thanks for your help!

**Comment by matthewmueller** [comment]
added travis and coveralls to this project :-)

**Comment by kichooo** [comment]
Nice!

**Comment by kichooo** [comment]
Wait, so you just removed all semicolons and called it a refactor? :S

File: lib/resolve.js [hunk 1]
```diff
@@ -101,35 +102,34 @@ function select($, selector) {
...
     default:
-      return $el.attr(attr);
+      return $el.attr(attr)
   }
 }
 
-
-// var cheerio = require('cheerio');
-// var $ = cheerio.load('<ul title="some title"><li>a</li><li>b</li></ul><ul><li>c</li><li>d</li></ul>');
+// var cheerio = require('cheerio')
```
Comment (syzer) [inline lib/resolve.js]: lol

**Comment by syzer** [comment]
@kichooo not only `;` , also ```function name(args)``` => ```function name (args)```

File: lib/resolve.js (hunk 1 above)
Comment (Kikobeats) [inline lib/resolve.js]: not me, a robot for me!

Read basic checklist related with [standard](https://github.com/feross/standard#rules) style.

**Pull request merged**